pkg/recipes/terraform/config/providers/types_test.go

```diff
@@ -94,12 +95,202 @@ func TestGetRecipeProviderConfigs(t *testing.T) {
 				},
 			},
 		},
+		{
+			desc: "provider with secrets",
+			envConfig: &recipes.Configuration{
+				RecipeConfig: datamodel.RecipeConfigProperties{
+					Terraform: datamodel.TerraformConfigProperties{
+						Providers: map[string][]datamodel.ProviderConfigProperties{
+							"azurerm": {
+								{
+									AdditionalProperties: map[string]any{
+										"subscriptionid": 1234,
+										"tenant_id":      "745fg88bf-86f1-41af-43ut",
+									},
+									Secrets: map[string]datamodel.SecretReference{
+										"secret1": {
+											Source: "secretstoreid1",
+											Key:    "secretkey1",
+										},
+									},
+								},
+								{
+									AdditionalProperties: map[string]any{
+										"alias":          "az-paymentservice",
+										"subscriptionid": 45678,
+										"tenant_id":      "gfhf45345-5d73-gh34-wh84",
+									},
+									Secrets: map[string]datamodel.SecretReference{
+										"secret2": {
+											Source: "secretstoreid2",
+											Key:    "secretkey2",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			secrets: map[string]map[string]string{
+				"secretstoreid1": {"secretkey1": "secretvalue1"},
+				"secretstoreid2": {"secretkey2": "secretvalue2"},
+			},
+			expected: map[string][]map[string]any{
+				"azurerm": {
+					{
+						"subscriptionid": 1234,
+						"tenant_id":      "745fg88bf-86f1-41af-43ut",
+						"secret1":        "secretvalue1",
+					},
+					{
+						"alias":          "az-paymentservice",
+						"subscriptionid": 45678,
+						"tenant_id":      "gfhf45345-5d73-gh34-wh84",
+						"secret2":        "secretvalue2",
+					},
+				},
+			},
+		},
+		{
+			desc: "provider and env with secrets",
+			envConfig: &recipes.Configuration{
+				RecipeConfig: datamodel.RecipeConfigProperties{
+					Terraform: datamodel.TerraformConfigProperties{
+						Providers: map[string][]datamodel.ProviderConfigProperties{
+							"azurerm": {
+								{
+									AdditionalProperties: map[string]any{
+										"subscriptionid": 1234,
+										"tenant_id":      "745fg88bf-86f1-41af-43ut",
+									},
+									Secrets: map[string]datamodel.SecretReference{
+										"secret1": {
+											Source: "secretstoreid1",
+											Key:    "secretkey1",
+										},
```

Comment:
can we also add a test convering: empty AdditionalProperties and non empty secrets